# **Fisik vs. Digital: Analisis Perbandingan Kerentanan Media Penyimpanan Data Kesehatan US**
#### Kelompok 2:
1. Adhitya Bayu Pratama   : 5027261045
2. Via Nur Belaya         : 5027261060
3. Raditya Langit Nugraha : 5027261072
## *Topik: Data Breaches*

Data ini sangat menarik untuk diteliti karena tidak hanya memetakan jenis-jenis media penyimpanan data kesehatan yang paling rentan mengalami kebocoran, tetapi juga mengungkap secara pasti seberapa besar jumlah individu yang dirugikan dari setiap insiden tersebut. Dari temuan tersebut, analisis ini disusun untuk menjawab beberapa pertanyaan yang muncul guna memahami pola, karakteristik, dan tingkat keparahan risiko keamanan data medis secara menyeluruh yaitu:
- Berapa perbandingan proporsi dan frekuensi insiden kebocoran data antara media dokumen fisik (paper records) dan perangkat elektronik?
- Bagaimana distribusi jenis pelanggaran (Type of Breach) pada media fisik dibandingkan dengan digital?
- Tipe entitas kesehatan (Covered Entity Type) mana yang paling sering mengalami kebocoran data pada media dokumen fisik versus perangkat elektronik?

In [230]:
import pandas as pd
import numpy as np
import matplotlib as mt
import seaborn as sns
#Kumpulan library yang kita gunakan

print("Library berhasil dimasukan") #Memastikan library sudah ter execute atau ter import dengan menampilkan teks (yang berarti line diatasnya sudah berhasil di proses)

Library berhasil dimasukan


In [231]:
df = pd.read_csv('breach_report.csv', index_col=0) #Membaca file .csv yang kita mau proses
print("Jumlah baris, kolom:", df.shape) #Menyatakan ukuran dari data yang akan kita proses

pd.set_option('display.max_rows', None) #menampilkan semua data tanpa di limit

df.head()

Jumlah baris, kolom: (1654, 9)


,Name of Covered Entity,State,Covered Entity Type,Individuals Affected,Breach Submission Date,Type of Breach,Location of Breached Information,Business Associate Present,Web Description
index,,,,,,,,,
0,Brooke Army Medical Center,TX,Healthcare Provider,1000.0,10/21/09,Theft,Paper/Films,No,A binder containing the protected health infor...
1,"Mid America Kidney Stone Association, LLC",MO,Healthcare Provider,1000.0,10/28/09,Theft,Network Server,No,Five desktop computers containing unencrypted ...
2,Alaska Department of Health and Social Services,AK,Healthcare Provider,501.0,10/30/09,Theft,"Other, Other Portable Electronic Device",No,\N
3,Health Services for Children with Special Need...,DC,Health Plan,3800.0,11/17/09,Loss,Laptop,No,A laptop was lost by an employee while in tran...
4,"Mark D. Lurie, MD",CA,Healthcare Provider,5166.0,11/20/09,Theft,Desktop Computer,No,A shared Computer that was used for backup was...


Setiap baris menyatakan tiap rumah sakit yang terkena kebocoran data

In [232]:
df = df.drop(['Name of Covered Entity','Breach Submission Date','Business Associate Present','Web Description'], axis=1)

## Deskripsi Data dan Kamus Data

### Sumber Data dan Lisensi

**License:** [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)

**You are free to:**
- **Share** — copy and redistribute the material in any medium or format for any purpose, even commercially.
- **Adapt** — remix, transform, and build upon the material for any purpose, even commercially.

**Under the following terms:**
- **Attribution** — You must give appropriate credit, provide a link to the license, and indicate if changes were made. You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.
- **No additional restrictions** — You may not apply legal terms or technological measures that legally restrict others from doing anything the license permits.

**Notices:**
- You do not have to comply with the license for elements of the material in the public domain or where your use is permitted by an applicable exception or limitation.
- No warranties are given. The license may not give you all of the permissions necessary for your intended use. For example, other rights such as publicity, privacy, or moral rights may limit how you use the material.

**Source:** [Kaggle - Major US Health Data Breaches](https://www.kaggle.com/datasets/thedevastator/major-us-health-data-breaches)

**Original source:**
- data.world (Web discontinued)
- [HHS OCR Breach Portal](https://ocrportal.hhs.gov/ocr/breach/breach_report_hip.jsf)

---

### Siapa yang Mengumpulkan Data? Kapan? Bagaimana?

**Siapa:**
Data ini dikumpulkan oleh *Office for Civil Rights* (OCR) di bawah naungan *U.S. Department of Health and Human Services* (HHS).

**Kapan:**
Data direkam secara berkelanjutan sejak *HIPAA Breach Notification Rule* mulai diberlakukan pada tahun 2009. Pengumpulannya dilakukan secara real-time, dan institusi wajib melaporkan insiden ke HHS selambat-lambatnya **60 hari** sejak kebocoran pertama kali ditemukan (bukan dihitung dari hari terjadinya kebocoran).

**Bagaimana:**
Data ini tidak dikumpulkan melalui survei atau pencarian independen oleh peneliti, melainkan lewat **mandat hukum wajib lapor** (HIPAA/HITECH Act).

---

### Jumlah Baris dan Kolom

Jumlah baris = 1654

Jumlah kolom = 9

### Kamus Data

*(Setelah mengecek kualitas data kami hanya akan memakai 5 kolom yang sesuai dengan tujuan analisa kami, tabel kamus data untuk kelima kolom yang dipakai akan dilengkapi di bawah)*

In [233]:
kamus_data = {
    "Kolom": [
        "Name of Covered Entity",
        "Covered Entity Type",
        "Individuals Affected",
        "Type of Breach",
        "Location of Breached Information"
    ],
    "Arti": [
        "The name of the entity (US Hospital) that experienced the data breach. (String)",
        "The type of entity that experienced the data breach, such as hospitals or insurers. (String)",
        "Individuals effected",
        "The type of breach that occurred, such as unauthorized access/disclosure or loss of hardware/electronic media. (String)",
        "The location of the breached information, such as paper records or electronic devices. (String)"
    ],
    "Jenis_variabel": [
        "Kualitatif",
        "Kualitatif",
        "Kuantitatif diskrit",
        "Kualitatif",
        "Kualitatif"
    ],
    "Skala": [
        "Nominal",
        "Nominal",
        "Ratio",
        "Nominal",
        "Nominal"
    ],
    "Satuan": ["-", "-", "-", "-", "-"],
    "Contoh": [
        "Brooke Army Medical Center",
        "Healthcare Provider",
        "1000",
        "Theft",
        "Paper/Films"
    ]
}

dtf = pd.DataFrame(kamus_data)
dtf.index += 1
dtf

,Kolom,Arti,Jenis_variabel,Skala,Satuan,Contoh
1,Name of Covered Entity,The name of the entity (US Hospital) that expe...,Kualitatif,Nominal,-,Brooke Army Medical Center
2,Covered Entity Type,The type of entity that experienced the data b...,Kualitatif,Nominal,-,Healthcare Provider
3,Individuals Affected,Individuals effected,Kuantitatif diskrit,Ratio,-,1000
4,Type of Breach,"The type of breach that occurred, such as unau...",Kualitatif,Nominal,-,Theft
5,Location of Breached Information,"The location of the breached information, such...",Kualitatif,Nominal,-,Paper/Films


In [234]:
df.info() #Pengecekan tipe data setiap kolom

<class 'pandas.DataFrame'>
RangeIndex: 1654 entries, 0 to 1653
Data columns (total 5 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   State                             1645 non-null   str    
 1   Covered Entity Type               1613 non-null   str    
 2   Individuals Affected              1632 non-null   float64
 3   Type of Breach                    1640 non-null   str    
 4   Location of Breached Information  1643 non-null   str    
dtypes: float64(1), str(4)
memory usage: 64.7 KB


In [235]:
df.isna().sum()  #Pengecekan data yang kosong di setiap kolomnya

State                                9
Covered Entity Type                 41
Individuals Affected                22
Type of Breach                      14
Location of Breached Information    11
dtype: int64

In [236]:
print("Baris yang terduplikat ada:",df.duplicated().sum(), "baris") #Pengecekan jumlah baris yang duplikat

Baris yang terduplikat ada: 12 baris


In [237]:
df[df.duplicated(keep=False)]

,State,Covered Entity Type,Individuals Affected,Type of Breach,Location of Breached Information
index,,,,,
162,CA,Business Associate,1474.0,Theft,"Other, Other Portable Electronic Device"
163,CA,Business Associate,1474.0,Theft,"Other, Other Portable Electronic Device"
173,TX,Healthcare Provider,585.0,Theft,Laptop
275,TX,Healthcare Provider,1500.0,Theft,Laptop
321,IL,Health Plan,501.0,Theft,Paper/Films
428,CA,NaN,NaN,NaN,NaN
511,TX,Healthcare Provider,585.0,Theft,Laptop
512,IL,Healthcare Provider,1500.0,Theft,Paper/Films
558,IL,Healthcare Provider,1500.0,Theft,Paper/Films


In [238]:
df = df.drop_duplicates()

In [239]:
df = df.dropna() #Membersihkan setiap baris yang tidak memiliki data di suatu kolom

In [240]:
df.isna().sum()

State                               0
Covered Entity Type                 0
Individuals Affected                0
Type of Breach                      0
Location of Breached Information    0
dtype: int64

Semua data sudah bersih, jadi pengolahan selanjutnya dapat dilakukan tanpa adanya baris yang tidak memiliki data di salah satu kolomnya.

In [241]:
df.head()

,State,Covered Entity Type,Individuals Affected,Type of Breach,Location of Breached Information
index,,,,,
0,TX,Healthcare Provider,1000.0,Theft,Paper/Films
1,MO,Healthcare Provider,1000.0,Theft,Network Server
2,AK,Healthcare Provider,501.0,Theft,"Other, Other Portable Electronic Device"
3,DC,Health Plan,3800.0,Loss,Laptop
4,CA,Healthcare Provider,5166.0,Theft,Desktop Computer


In [242]:
col_to_split = 'Type of Breach'
col_to_split1 = 'Location of Breached Information'

df[col_to_split] = df[col_to_split].astype(str).str.split(', ')
df[col_to_split1] = df[col_to_split1].astype(str).str.split(', ')

df_exploded = df.explode(col_to_split).explode(col_to_split1).reset_index(drop=True)
df_exploded.head()

,State,Covered Entity Type,Individuals Affected,Type of Breach,Location of Breached Information
0,TX,Healthcare Provider,1000.0,Theft,Paper/Films
1,MO,Healthcare Provider,1000.0,Theft,Network Server
2,AK,Healthcare Provider,501.0,Theft,Other
3,AK,Healthcare Provider,501.0,Theft,Other Portable Electronic Device
4,DC,Health Plan,3800.0,Loss,Laptop


In [244]:
frekuensi = df_exploded["Covered Entity Type"].value_counts()
persentase = df_exploded["Covered Entity Type"].value_counts(normalize=True) * 100

# Digabung jadi satu tabel biar rapih
ringkasan = pd.DataFrame({
    "Frekuensi": frekuensi,
    "Persentase (%)": persentase.round(2)
})
print(ringkasan)

modus = df_exploded["Covered Entity Type"].mode()[0]
print(f"\nData yang paling sering muncul: {modus}")

                           Frekuensi  Persentase (%)
Covered Entity Type                                 
Healthcare Provider             1387           70.33
Business Associate               368           18.66
Health Plan                      213           10.80
Healthcare Clearing House          4            0.20

Data yang paling sering muncul: Healthcare Provider
